pkg/client/client.go

```diff
@@ -148,28 +142,45 @@ func newKubeClient(config *restclient.Config) (controllercli.Client, error) {
 }
 
 func buildKubeConnectionConfig() (*restclient.Config, error) {
-	config, err := clientcmd.BuildConfigFromFlags("", getKubeConfigFile())
+	config, err := clientcmd.BuildConfigFromFlags("", GetKubeConfigFile())
 	if err != nil {
 		return nil, err
 	}
 	return config, nil
 }
 
-func getKubeConfigFile() string {
-	kubeconfig := util.GetOSEnv(envVarKubeConfig, "")
-	if len(kubeconfig) > 0 {
-		log.Debugf("Kube config file read from %s environment variable: %s", envVarKubeConfig, kubeconfig)
-		return kubeconfig
+// GetKubeConfigFile gets the .kubeconfig file.
+// Never returns an empty string, fallback to default path if not present in the known locations
+func GetKubeConfigFile() string {
+	filename := clientcmd.NewDefaultPathOptions().GetDefaultFilename()
+	// make sure the path to the file exists
+	if _, err := os.Stat(filename); os.IsNotExist(err) {
+		dirName := filepath.Dir(filename)
+		if err := os.MkdirAll(dirName, os.ModePerm); err != nil {
+			panic(fmt.Errorf("Error while trying to create kube config directories %s: %s ", filename, err))
+		}
+	} else if err != nil {
+		panic(fmt.Errorf("Error while trying to access the kube config file %s: %s ", filename, err))
 	}
-	log.Debug("Trying to get kube config file from HOME directory")
-	if home := util.GetHomeDir(); home != "" {
-		kubeconfig = filepath.Join(home, defaultKubeConfigPath)
+	if file, err := os.OpenFile(filename, os.O_RDONLY|os.O_CREATE, 0666); err != nil {
```

Comment:
is there a need to specify filemask while opening a file?

Reply:
yes, to guarantee that the file is there. If a user open the CLI without a kubeconfig we can panic with a well defined message.

Reply:
I thought that it is not necessary, even if set, it would be ignored, did you get any issue if not setting it?
666 seems not to be the correct permissions for kube/config file though, i guess it is 600.

Reply:
I'm using the same one as the oc client.

Reply:
mine is 600

Reply:
mine too is 600

Reply:
I'll change it. Thanks guys!